django/contrib/admin/templates/admin/login.html

```diff
@@ -30,7 +30,16 @@
 {% endfor %}
 {% endif %}
 
-<div id="content-main">
+<p id="content-main">
+
+{% if user.is_authenticated %}
```

Comment:
I don't think it's guaranteed that users viewing the login page while they are authenticated will have been redirected there.